src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -23,11 +23,11 @@ public class NoOpRestoreUtilities
         internal const string NoOpCacheFileName = "project.nuget.cache";
 
         /// <summary>
-        /// If the dependencyGraphSpec is not set, we cannot no-op on this project restore. 
+        /// If the dependencyGraphSpec is not set or <see cref="RestoreRequest.AllowNoOp" /> is <code>false</code>, we cannot no-op on this project restore.
         /// </summary>
         internal static bool IsNoOpSupported(RestoreRequest request)
         {
-            return request.DependencyGraphSpec != null;
+            return request.DependencyGraphSpec != null && request.AllowNoOp;
```

Comment:
Can you please add a quick test for this? 

https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs is the right place I believe.

Reply:
Added!